lib/spack/spack/build_environment.py

```diff
@@ -781,42 +724,39 @@ def load_external_modules(pkg):
 
 def setup_package(pkg, dirty, context='build'):
     """Execute all environment setup routines."""
+    if context not in ['build', 'test']:
+        raise ValueError(
+            "'context' must be one of ['build', 'test'] - got: {0}"
+            .format(context))
+
+    set_module_variables_for_package(pkg)
+
     env = EnvironmentModifications()
 
     if not dirty:
         clean_environment()
 
-    # setup compilers and build tools for build contexts
+    # setup compilers for build contexts
     need_compiler = context == 'build' or (context == 'test' and
                                            pkg.test_requires_compiler)
     if need_compiler:
         set_compiler_environment_variables(pkg, env)
-        set_build_environment_variables(pkg, env, dirty)
+        set_wrapper_variables(pkg, env)
+
+    env.extend(modifications_from_dependencies(pkg.spec, context))
 
     # architecture specific setup
     pkg.architecture.platform.setup_platform_environment(pkg, env)
 
-    if context == 'build':
-        # recursive post-order dependency information
-        env.extend(
-            modifications_from_dependencies(pkg.spec, context=context)
-        )
-
-        if (not dirty) and (not env.is_unset('CPATH')):
-            tty.debug("A dependency has updated CPATH, this may lead pkg-"
-                      "config to assume that the package is part of the system"
-                      " includes and omit it when invoked with '--cflags'.")
+    if (not dirty) and (not env.is_unset('CPATH')):
+        tty.debug("A dependency has updated CPATH, this may lead pkg-"
+                  "config to assume that the package is part of the system"
+                  " includes and omit it when invoked with '--cflags'.")
 
-        # setup package itself
-        set_module_variables_for_package(pkg)
+    if context == 'build':
         pkg.setup_build_environment(env)
     elif context == 'test':
-        import spack.user_environment as uenv  # avoid circular import
-        env.extend(uenv.environment_modifications_for_spec(pkg.spec))
```

Comment:
This has been removed, there was some overlap between this function and `uenv.environment_modifications_for_spec`: everything there happens here now except

```
    env = environment.inspect_path(
        spec.prefix,
        prefix_inspections(spec.platform),
        exclude=environment.is_system_path
    )
```

which is not considered essential for this function.